pyvista/core/filters/image_data.py

```diff
@@ -1604,3 +1607,280 @@ def _update_and_get_output():
+
+        >>> points = [(2, 1, 0), (0, 0, 1)]
+        >>> connected = segmented_grid.label_connectivity(
+        ...     extraction_mode='seeded', point_seeds=points
+        ... )
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        """
+        # Get a copy of input to not overwrite data
+        input_mesh = self.copy()  # type: ignore[attr-defined]
+
+        if scalars is None:
+            set_default_active_scalars(input_mesh)
+        else:
+            input_mesh.set_active_scalars(scalars)
+        # Make sure we have point data (required by the filter)
+        field, scalars = input_mesh.active_scalars_info
+
+        if field == FieldAssociation.CELL:
+            # Convert to point data
+            input_mesh = input_mesh.cells_to_points(scalars=scalars, copy=False)
+
+        # Set vtk algorithm
+        alg = _vtk.vtkImageConnectivityFilter()
+        alg.SetInputDataObject(input_mesh)
+        alg.SetInputArrayToProcess(
+            0,
+            0,
+            0,
+            field.value,
+            scalars,
+        )  # args: (idx, port, connection, field, name)
+
+        # Set the scalar range considered for connectivity
+        # Default to vtk default 0.5 to VTK_DOUBLE_MAX, nothing to do
+        # See https://vtk.org/doc/nightly/html/classvtkImageConnectivityFilter.html
+        if scalar_range is not None:
+            if isinstance(scalar_range, str) and scalar_range == 'auto':
+                unique_scalars = np.unique(input_mesh.point_data[scalars])
+                scalar_range = [unique_scalars[1], unique_scalars[-1]]
+            else:
+                scalar_range = _validation.validate_data_range(scalar_range)
+            alg.SetScalarRange(*scalar_range)
+
+        if extraction_mode == 'all':
+            alg.SetExtractionModeToAllRegions()
+        elif extraction_mode == 'largest':
+            alg.SetExtractionModeToLargestRegion()
+        elif extraction_mode == 'seeded':
+            if point_seeds is None:
+                raise ValueError(
+                    '`point_seeds` must be specified when `extraction_mode="point_seeds"`.',
+                )
+            elif isinstance(point_seeds, (Sequence, np.ndarray)):
+                # Ensure points are floats, see pyvista.core.utilities.points
+                point_seeds = np.asanyarray(point_seeds, dtype=float)
+                # PointSet requires vtk >= 9.1.0
+                # See https://docs.pyvista.org/api/core/_autosummary/pyvista.pointset#pyvista.PointSet
+                if pyvista.vtk_version_info >= (9, 1, 0):
+                    point_seeds = pyvista.PointSet(point_seeds)
+                else:
+                    point_seeds = pyvista.PolyData(point_seeds)
```

Comment:
The DataSet constructors automatically cast to float32 and emit a warning if points are integers, maybe just let the constructor handle this conversion? And if PointSet is not specifically required by the filter, maybe just use PolyData (one less branch to test)?
```suggestion
                point_seeds = pyvista.PolyData(point_seeds)
```

Reply:
I simplified the logic as suggested but did not remove PointSet for now according to the above (incomplete) data.

Reply:
Makes sense, yes it seems `PointSet` is better to use generally if cells are not required. As a side note, it's possible to avoid creating cells with `PolyData` by assigning points outside of the constructer, e.g. `p = pv.PolyData(); p.points = points`. Running `timeit` as above but with this change yields similar performance to using `PointSet`. Either way, using `PointSet` is fine, I think I saw missing code coverage here so that was part of the motivation for this change. I'll mark this as resolved.